infra/bridge/src/utils/exchange-rate.js

```diff
@@ -5,49 +5,80 @@ const request = require('superagent')
 const { getAsync, redisClient } = require('../utils/redis')
 const logger = require('../logger')
 
+const currencies = [
+  'ETH-USD',
+  // DAI is disabled because coingecko doesn't support it
+  // however DAI is pegged to USD so maybe we could just use 1
+  // or call a different API to get it. Currently pegged to 1 in
+  // search.js
+  // 'DAI-USD',
+  'KRW-USD',
+  'SGD-USD',
+  'GBP-USD',
+  'EUR-USD',
+  'JPY-USD',
+  'CNY-USD'
+]
+
 function startExchangeRatePolling() {
   // TODO: Store the markets to be polled somewhere or in ENV and start poll with that.
-  pollExchangeRate('ETH-USD')
-  pollExchangeRate('DAI-USD')
+  pollExchangeRate(currencies)
 }
 
 /**
  * Recursively polls the exchange rate in fixed intervals
  */
-async function pollExchangeRate(market) {
+async function pollExchangeRate(markets) {
   setTimeout(() => {
-    fetchExchangeRate(market).then(() => pollExchangeRate(market))
+    fetchExchangeRate(markets).then(() => pollExchangeRate(markets))
   }, process.env.EXCHANGE_RATE_POLL_INTERVAL || 30000)
 }
 
 /**
- * Fetch excahnge rate from remote URL
+ * Fetch exchange rate from coingecko
```

Comment:
I grepped our code base to find out what component was using the bridge /utils/exchange-rate endpoint. It looks like the mobile app is.
See here: https://github.com/OriginProtocol/origin/blob/master/mobile/src/screens/marketplace.js#L564
Note that it queries for the exchange rate for ETH and DAI against the user's preferred fiat currency (e.g. USD, EUR, GBP, etc...)

So we need to support DAI exchange rates otherwise it will break the mobile app...

Either you could add some code to calculate those DAI exchange rates assuming thet DAI-USD is 1 or we stick to using cryptonator ?

Reply:
Awesome catch sir @franckc 🎉 As Frank said this has to support all the same functionality that was here before.

Reply:
Ah yes good catch! I had implemented the hardcoding of DAI-USD in search.js but I can move it here and store DAI-USD as '1' in redis. I'd prefer to stick to one API as they all return different formats and it makes for a lot messier code. Unless there's any objections to pegging DAI-USD to '1'?

Reply:
@franckc I ended up adding a statement to use the USD value, instead of 1, whenever DAI is involved. Since it is pegged to USD this should be a better approach.

[All the exchange rates are calculated based on values returned from coingecko.](https://github.com/OriginProtocol/origin/blob/f2f2c0b6240b3e5550d601a5953fa3a7002ac3bc/infra/bridge/src/utils/exchange-rate.js#L67-L85) so this shouldn't be an issue.

Reply:
Just noticed an error in my comment above the rate calculation (see the link above), the value is no longer calculated against just USD, its calculated against whatever the exchangeToSymbol is set as.